lib/chef/resource/windows_task.rb

```diff
@@ -136,65 +154,72 @@ def password_required?(user)
-              raise ArgumentError, "frequency_modifier value #{frequency_modifier} is invalid. Valid values for :weekly frequency are 1 - 52."
+        if ([:on_logon, :onstart, :on_idle, :none].include?(frequency)) && ( frequency_modifier != 1)
+          raise ArgumentError, "frequency_modifier property not supported with frequency :#{frequency}"
+        end
+
+        if frequency == :monthly
+          unless ("1".."12").include?(frequency_modifier.to_s.upcase) || frequency_modifier_includes_days_of_weeks?(frequency_modifier)
+            raise ArgumentError, "frequency_modifier value #{frequency_modifier} is invalid. Valid values for :monthly frequency are 1 - 12, 'FIRST', 'SECOND', 'THIRD', 'FOURTH', 'LAST', 'LASTDAY'."
+          end
+        else
+          unless frequency.nil? || frequency_modifier.nil?
+            frequency_modifier = frequency_modifier.to_i
+            min = max = 1
+            case frequency
+              when :minute
+                max = 1439
+              when :hourly
+                max = 23
+              when :daily
+                max = 365
+              when :weekly
+                max = 52
             end
-          when :monthly
-            unless ("1".."12").to_a.push("FIRST", "SECOND", "THIRD", "FOURTH", "LAST", "LASTDAY").include?(frequency_modifier.to_s.upcase)
-              raise ArgumentError, "frequency_modifier value #{frequency_modifier} is invalid. Valid values for :monthly frequency are 1 - 12, 'FIRST', 'SECOND', 'THIRD', 'FOURTH', 'LAST', 'LASTDAY'."
+            unless frequency_modifier >= min && frequency_modifier <= max
+              raise ArgumentError, "frequency_modifier value #{frequency_modifier} is invalid. Valid values for :#{frequency} frequency are #{min} - #{max}."
             end
           end
         end
       end
 
-      def validate_create_day(day, frequency)
-        unless [:weekly, :monthly].include?(frequency)
-          raise "day property is only valid for tasks that run monthly or weekly"
-        end
+      def validate_create_day(day, frequency, frequency_modifier)
+        raise "day property is only valid for tasks that run monthly or weekly" unless [:weekly, :monthly].include?(frequency)
+
         if day.is_a?(String) && day.to_i.to_s != day
           days = day.split(",")
-          days.each do |d|
-            unless ["mon", "tue", "wed", "thu", "fri", "sat", "sun", "*"].include?(d.strip.downcase)
-              raise ArgumentError, "day property invalid. Only valid values are: MON, TUE, WED, THU, FRI, SAT, SUN and *. Multiple values must be separated by a comma."
+          if days_includes_days_of_months?(days)
+              raise "day values 1-31 or last is only valid with frequency :monthly" if frequency == :weekly
+              raise "Do not use frequency_modifier property when day values set as 1-31 or last" if frequency_modifier && frequency_modifier != 1
```

Comment:
We should add tests for these two raises given the complexity the conditional logic. If they are covered, it is difficult to correlate and understand due to the complexity.

Reply:
@rhass for failing test cases I hope you are having right build of 'win32-taskscheduler'. We have fixed some issues in win32-taskcheduler and window_task now depends on this gem. And the new release is not yet done for win32-taskschduler so on my local in my Gemfile so you might need to point to win32-taskscheduler ole branch on github in Gemfile and comment out dependency line for win32-taskscheduler library in chef-universal-mingw32.gemspec.

For adding tests for raise error I will work on it.